client/application/home/home.controller.js

```diff
@@ -4,23 +4,158 @@
 
   angular.module('caproomster').controller('HomeController', HomeController);
 
-  HomeController.$inject = ['$scope', '$state'];
-
-  function HomeController($scope, $state) {
-
-    $scope.uiConfig = {
-      calendar:{
-        height: 450,
-        editable: true,
-        header:{
-          left: 'month basicWeek basicDay agendaWeek agendaDay',
-          center: 'title',
-          right: 'today prev,next'
-        },
-        eventClick: $scope.alertEventOnClick,
-        eventDrop: $scope.alertOnDrop,
-        eventResize: $scope.alertOnResize
+  HomeController.$inject = ['$scope', 'moment', 'calendarConfig'];
+
+  function HomeController($scope, moment, calendarConfig) {
+
+    // Init controller and variables
+    init();
+    $scope.toggleMenu = toggleMenu;
+    $scope.changeRoom = changeRoom;
+    $scope.addEvent = addEvent;
+    $scope.eventClicked = eventClicked;
+    $scope.eventEdited = eventEdited;
+    $scope.eventDeleted = eventDeleted;
+    $scope.eventTimesChanged = eventTimesChanged;
+    $scope.timespanClicked = timespanClicked;
+
+    // Init function
+
+    function init() {
+      $scope.calendarView = 'week';
+      $scope.viewDate = new Date();
+      $scope.toggleText = "Show Room List";
+      $scope.roomNumber = 'H921';
+      $scope.cellIsOpen = true;
+      $scope.events = [];
+      fetchEvents();
+    }
+
+    // Toggle Menu
+
+    function toggleMenu() {
+      if (!$scope.isToggled) {
+        $scope.isToggled = true;
+        $scope.toggleText = "Hide Room List";
+      } else {
+        $scope.isToggled = false;
+        $scope.toggleText = "Show Room List";
       }
+    }
+
+    // Change room number and fetch room data
+
+    function changeRoom(room) {
+      $scope.roomNumber = room;
+      // TODO: change view, get data etc.
+    }
+
+    // Fetch reservations from backend
+
+    function fetchEvents() {
+      // TODO: integrate rest api and fecth real date, for now just mocking
+      var actions = [{
+        label: '<i class=\'glyphicon glyphicon-pencil\'></i>',
+        onClick: function(args) {
+          console.log('Edited');
+        }
+      }, {
+        label: '<i class=\'glyphicon glyphicon-remove\'></i>',
+        onClick: function(args) {
+          console.log('Deleted');
+        }
+      }];
+      // mock data
+      $scope.events = [
+        {
```

Comment:
This library requires specific format of event, therefore whatever the data format is in the core we will need to parse it on front-end